kernel/src/log_segment.rs

```diff
@@ -81,28 +81,49 @@ impl LogSegment {
         }
 
         // Get the effective version from chosen files
-        let version_eff = ascending_commit_files
+        let effective_version = ascending_commit_files
             .last()
             .or(checkpoint_parts.first())
-            .ok_or(Error::generic("No files in log segment"))?
+            .ok_or(Error::EmptyLogSegment)?
             .version;
         if let Some(end_version) = end_version {
             require!(
-                version_eff == end_version,
+                effective_version == end_version,
                 Error::generic(format!(
                     "LogSegment end version {} not the same as the specified end version {}",
-                    version_eff, end_version
+                    effective_version, end_version
                 ))
             );
         }
+
         Ok(LogSegment {
-            end_version: version_eff,
+            end_version: effective_version,
             log_root,
             ascending_commit_files,
             checkpoint_parts,
         })
     }
 
+    /// concatenate two LogSegments (in order). The new LogSegment must start at the
+    /// (end version + 1) of the existing LogSegment. And the new LogSegment must have the same log
+    /// root.
+    pub(crate) fn try_concat(mut front: LogSegment, back: LogSegment) -> DeltaResult<LogSegment> {
```

Comment:
I think I would prefer old/new or earlier/later, or something that tells me the order. front/back isn't clear.

But actually why not `try_append(&self, segment: LogSegment)`, which would append the segment to myself and return a new segment?